tokens/src/mock.rs

```diff
@@ -89,6 +98,186 @@ impl OnReceived<AccountId, CurrencyId, Balance> for MockOnReceived {
+	fn convert(x: u128) -> u64 {
+		x as u64
+	}
+}
+
+parameter_types! {
+	pub const CandidacyBond: u64 = 3;
+}
+
+thread_local! {
+	static VOTING_BOND: RefCell<u64> = RefCell::new(2);
+	static DESIRED_MEMBERS: RefCell<u32> = RefCell::new(2);
+	static DESIRED_RUNNERS_UP: RefCell<u32> = RefCell::new(2);
+	static TERM_DURATION: RefCell<u64> = RefCell::new(5);
+}
+
+pub struct VotingBond;
+impl Get<u64> for VotingBond {
+	fn get() -> u64 {
+		VOTING_BOND.with(|v| *v.borrow())
+	}
+}
+
+pub struct DesiredMembers;
+impl Get<u32> for DesiredMembers {
+	fn get() -> u32 {
+		DESIRED_MEMBERS.with(|v| *v.borrow())
+	}
+}
+
+pub struct DesiredRunnersUp;
+impl Get<u32> for DesiredRunnersUp {
+	fn get() -> u32 {
+		DESIRED_RUNNERS_UP.with(|v| *v.borrow())
+	}
+}
+
+pub struct TermDuration;
+impl Get<u64> for TermDuration {
+	fn get() -> u64 {
+		TERM_DURATION.with(|v| *v.borrow())
+	}
+}
+
+thread_local! {
+	pub static MEMBERS: RefCell<Vec<u64>> = RefCell::new(vec![]);
+	pub static PRIME: RefCell<Option<u64>> = RefCell::new(None);
+}
+
+pub struct TestChangeMembers;
+impl ChangeMembers<u64> for TestChangeMembers {
+	fn change_members_sorted(incoming: &[u64], outgoing: &[u64], new: &[u64]) {
+		// new, incoming, outgoing must be sorted.
+		let mut new_sorted = new.to_vec();
+		new_sorted.sort();
+		assert_eq!(new, &new_sorted[..]);
+
+		let mut incoming_sorted = incoming.to_vec();
+		incoming_sorted.sort();
+		assert_eq!(incoming, &incoming_sorted[..]);
+
+		let mut outgoing_sorted = outgoing.to_vec();
+		outgoing_sorted.sort();
+		assert_eq!(outgoing, &outgoing_sorted[..]);
+
+		// incoming and outgoing must be disjoint
+		for x in incoming.iter() {
+			assert!(outgoing.binary_search(x).is_err());
+		}
+
+		let mut old_plus_incoming = MEMBERS.with(|m| m.borrow().to_vec());
+		old_plus_incoming.extend_from_slice(incoming);
+		old_plus_incoming.sort();
+
+		let mut new_plus_outgoing = new.to_vec();
+		new_plus_outgoing.extend_from_slice(outgoing);
+		new_plus_outgoing.sort();
+
+		assert_eq!(
+			old_plus_incoming, new_plus_outgoing,
+			"change members call is incorrect!"
+		);
+
+		MEMBERS.with(|m| *m.borrow_mut() = new.to_vec());
+		PRIME.with(|p| *p.borrow_mut() = None);
+	}
+
+	fn set_prime(who: Option<u64>) {
+		PRIME.with(|p| *p.borrow_mut() = who);
+	}
+}
+
+parameter_types! {
+	pub const ElectionsPhragmenModuleId: LockIdentifier = *b"phrelect";
+}
+
+impl pallet_elections_phragmen::Trait for Runtime {
+	type ModuleId = ElectionsPhragmenModuleId;
+	type Event = TestEvent;
+	type Currency = CurrencyAdapter<Runtime, GetTokenId>;
+	type CurrencyToVote = CurrencyToVoteHandler;
+	type ChangeMembers = TestChangeMembers;
+	type InitializeMembers = ();
+	type CandidacyBond = CandidacyBond;
+	type VotingBond = VotingBond;
+	type TermDuration = TermDuration;
+	type DesiredMembers = DesiredMembers;
+	type DesiredRunnersUp = DesiredRunnersUp;
+	type LoserCandidate = ();
+	type KickedMember = ();
+	type BadReport = ();
+	type WeightInfo = ();
+}
```

Comment:
Maybe I missed something, but didn't see any usage for the new added mocks.

Reply:
Because `pallet_elections_phragmen` and `pallet_treasury` module dispatch is private function. Maybe I can test it from genesis config.